frontend/packages/console-dynamic-plugin-sdk/src/utils/rbac.tsx

```diff
@@ -0,0 +1,202 @@
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?,
+): [boolean, boolean] => {
+  const [loading, setLoading] = useSafetyFirst(true);
+  const [isAllowed, setAllowed] = useSafetyFirst(false);
+  // Destructure the attributes to pass them as dependencies to `useEffect`,
+  // which doesn't do deep comparison of object dependencies.
+  const {
+    group = '',
+    resource = '',
+    subresource = '',
+    verb = '' as K8sVerb,
+    name = '',
+    namespace = '',
+  } = resourceAttributes;
+  const impersonateKey = getImpersonateKey(impersonate);
+  React.useEffect(() => {
+    checkAccessInternal(group, resource, subresource, verb, name, namespace, impersonateKey)
+      .then((result: SelfSubjectAccessReviewKind) => {
+        setAllowed(result.status.allowed);
+        setLoading(false);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('SelfSubjectAccessReview failed', e);
+        // Default to enabling the action if the access review fails so that we
+        // don't incorrectly block users from actions they can perform. The server
+        // still enforces access control.
+        setAllowed(true);
+        setLoading(false);
+      });
+  }, [setLoading, setAllowed, group, resource, subresource, verb, name, namespace, impersonateKey]);
+
+  return [isAllowed, loading];
+};
+
+export const useMultipleAccessReviews = (
+  multipleResourceAttributes: AccessReviewResourceAttributes[],
+  impersonate?: boolean,
+): [AccessReviewsResult[], boolean] => {
+  const [loading, setLoading] = React.useState(true);
+  const [allowedArr, setAllowedArr] = React.useState<AccessReviewsResult[]>([]);
+
+  React.useEffect(() => {
+    const promises = multipleResourceAttributes.map((resourceAttributes) =>
+      checkAccess(resourceAttributes, impersonate),
+    );
+
+    Promise.all(promises)
+      .then((values) => {
+        setLoading(false);
+        const updatedAllowedArr = values.map<AccessReviewsResult>((result) => ({
+          resourceAttributes: result.spec.resourceAttributes,
+          allowed: result.status.allowed,
+        }));
+        setAllowedArr(updatedAllowedArr);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('SelfSubjectAccessReview failed', e);
+        setLoading(false);
+      });
+  }, [impersonate, multipleResourceAttributes]);
+
+  return [allowedArr, loading];
+};
+
+const RequireCreatePermission_: React.FC<RequireCreatePermissionProps> = ({
+  model,
+  namespace,
+  impersonate,
+  children,
+}) => {
+  const [isAllowed] = useAccessReview(
+    {
+      group: model?.apiGroup,
+      resource: model?.plural,
+      verb: 'create',
+      namespace,
+    },
+    impersonate,
+  );
+  return isAllowed ? <>{children}</> : null;
+};
+export const RequireCreatePermission = connect(impersonateStateToProps)(RequireCreatePermission_);
+RequireCreatePermission.displayName = 'RequireCreatePermission';
```

Comment:
Why is this (along with all related types) still here in the dynamic-plugin-sdk?

`RequireCreatePermission` isn't good for the API because it's too specific to create. However a more generic component would be ok. Let's omit this from the API for now.